### PR TITLE
Add pre-commit autoupdate workflow

### PR DESCRIPTION
**Pull request by altheaden**
This adds a GitHub actions workflow that will run the `pre-commit autoupdate` command to update pre-commit and its dependencies. The workflow is set to run weekly on Wednesdays.

Checklist
* [x] `Testing` comment in the PR documents testing used to verify the changes

### CODE REVIEW

**Comment by altheaden** [comment]
@xylar Can you tell me if you see anything obvious that should be changed about this workflow?

File: .github/workflows/pre_commit_update_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Pre-commit auto-update
+
+on:
+  schedule:
+    - cron: '0 8 * * 3'
```
Comment (xylar) [inline .github/workflows/pre_commit_update_workflow.yml:5]: Could we add a comment to explain what this last line means?

**Comment by xylar** [comment]
Very nice work, by the way!  I'm looking forward to seeing this in action!

**Review by xylar — APPROVED** [review]
Looks superb!

**Pull request merged**